apache/superset
# feat(native-filters): Adhoc dashboard native filters

### SUMMARY

Our users really like the functionality of the adhoc filters in the explore view and were interested in something similar for for dashboard filtering. Anything that can be done with an adhoc filter native filter can be done with the other filter types and vice versa, but in order to filter by any given column the user has to either:
- create a new filter (which can be a hassle to do on an adhoc basis),
- or have a value filter for every column in the dashboard which is not feasible in every dashboard.

In this PR we added a new native filter type that reuses the adhoc filter component from the explore view to allow for more complex filtering to easily be done in a dashboard.

This solution provides the user with a lot of power and flexibility in dashboards, which is why it is behind a feature flag for now. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


https://user-images.githubusercontent.com/102618419/202734225-c44cfa32-40e5-4edd-8cab-83dafe94bb2a.mp4

https://user-images.githubusercontent.com/102618419/202734246-d650a4e7-c964-4f07-8a31-ea6e0e907270.mp4

### TESTING INSTRUCTIONS

- Enable Feature Flag: ADHOC_DASHBOARD_NATIVE_FILTERS
- Navigate to dashboard
- Add/Edit Filters on the sidebar
- A new filter type should appear called "Adhoc"

### ADDITIONAL INFORMATION


- [x] Required feature flags: ADHOC_DASHBOARD_NATIVE_FILTERS
